lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc

```diff
@@ -135,15 +142,17 @@ function dosomething_northstar_openid_connect_post_authorize($tokens, $account,
  * Save a post-authorization destination in the session (e.g. for signing
  * up for a campaign).
  *
- * @param $action
- * @param $nid
+ * @param string $action
+ * @param string $nid
+ * @param array $param
  */
-function dosomething_northstar_set_authorization_action($action, $nid) {
+function dosomething_northstar_set_authorization_action($action, $nid, $params = []) {
   if ($action !== 'signup') {
     return;
   }
 
   $_SESSION['openid_connect_action'] = ['signup' => $nid];
+  $_SESSION['affiliate_messaging_opt_in'] = isset($params['affiliate_messaging_opt_in']) ? $params['affiliate_messaging_opt_in'] : FALSE;
```

Comment:
Ah, good call. For some reason I had it in my head that `$_SESSION['openid_connect_action'] = ['signup' => $nid];` needed to be separate. Ok, will update!